src/core/ts/background-scripts/background.js/telemetry/Telemetry.ts

```diff
@@ -47,6 +60,27 @@ export class Telemetry {
       console.info(
         `Telemetry: initialization completed with application ID ${appId}.`,
       );
+
+      // Submit queued pings every hour, or as dictated by telemetryDispatchIntervalInSecondsOverride
+      const periodInSeconds = telemetryDispatchIntervalInSecondsOverride
+        ? telemetryDispatchIntervalInSecondsOverride
+        : 60 * 60;
+      console.debug(
+        `Setting up the "${this.periodicAlarmName}" periodic callback to fire every ${periodInSeconds} seconds`,
+      );
+      const alarmListener = async alarm => {
```

Comment:
Right now, those pings gets lost. Need to persist the queue to avoid that.

Reply:
Note that there is no reliable "onShutdown" or browser close event (https://bugzilla.mozilla.org/show_bug.cgi?id=1606215). Persisting the queue in its current form is a bit tricky since it consists of callbacks. We'd need to make the ping-recordings serializable to properly be able to persist them.

Reply:
The purpose of this PR is to strike a temporary balance (eg given constraints of schema v1) between not sending pings all the time and risking to loose some data.

Here are some scenarios that currently leads to data loss:
A. User opens Nightly, tests translation and then closes the browser within one hour, returning to Nightly another day
B. User opens Nightly, tests translation and then closes the browser within one hour, never returning to Nightly
C. User opens Nightly, browses around without actually translating anything or interacting with the infobar and then closes the browser within one hour, returning to Nightly another day
D. User opens Nightly, browses around without actually translating anything or interacting with the infobar and then closes the browser within one hour, never returning to Nightly

Here are some up-until-now discussed solutions:
1. Persist queue across browser sessions
2. Submit at certain trigger events

Solution 1 leads to data loss in scenarios B and D above. 
Solution 2 leads to data loss in scenarios C and D above. 
A combination of solutions 1 and 2 leads to data loss in scenario D above.

If it is a hard requirement never to loose any data, not even in scenario D above, we need some other solution, one that we have not yet discussed. 

How about this: We decrease the default "one hour" interval down to something like 5 seconds, which makes data loss in any of the above scenarios very unlikely (a data loss window of 5 seconds seems reasonable), while still grouping many pings together. Optionally, we can even let the interval gradually increase as more time flows by, incrementally increasing the interval back to an hour or even a day/week etc for long term users of the browser.

Reply:
It shouldn't group many pings together. It should group interaction metrics and events for one page only. Other pages' interactions should be in other pings. We cannot lose any data when the user translates something, it should be submitted immediately. Timers can be used if we don't have any triggers, like scenarios when infobar is displayed and a user doesn't translate anything or language pair is not supported. It's ok to lose some inactivity data, but yes, 1 hour interval seems like too much for nightly. I would suggest reducing it to 1 min if we still want to use timer for inactivity events. This schema and implementation will live for a while, so too hacky and temporary implementations don't work for us.

Reply:
> It shouldn't group many pings together. It should group interaction metrics and events for one page only. Other pages' interactions should be in other pings. 

I was under the impression that data analysis would be just as easy when many pages' pings were grouped together. I think you are saying that this is not the case, so we need to only group pings within the same page's translation session to make it easier to query the collected data?

> We cannot lose any data when the user translates something, it should be submitted immediately. 

You propose/insist on having Solution 2 to be implemented.

> Timers can be used if we don't have any triggers, like scenarios when infobar is displayed and a user doesn't translate anything or language pair is not supported. It's ok to lose some inactivity data, but yes, 1 hour interval seems like too much for nightly. I would suggest reducing it to 1 min if we still want to use timer for inactivity events. 

Solution 3 as well. Using a 1 minute interval rather than 1 hour or 5 seconds as has been discussed above. Sure thing!

> This schema and implementation will live for a while, so too hacky and temporary implementations don't work for us.

With "temporary" I mean exactly that, eg that the implementation will only live for as long as schema v1 lives, something I thought was just a few weeks/months but it seems that you intend to keep it around for longer than that. 

Do we have a common understanding thus that we'll invest a bit more upfront and get a combination of solution 2 and 3 included as part of this PR, so that schema v1 can live for a longer period?